atomicapp/providers/openshift.py

```diff
@@ -133,6 +146,66 @@ def process_template(self, url, template):
             logger.error(msg)
             raise ProviderFailedException(msg)
 
+    def execute(self, namespace, pod, container, command,
+                outfile=None):
+        """
+        Execute a command in a container in an Openshift pod.
+
+        Args:
+            namespace (str): Namespace
+            pod (str): Pod name
+            container (str): Container name inside pod
+            command (str): Command to execute
+            outfile (str): Path to output file where results should be dumped
+
+        Returns:
+            Command output (str) or None in case results dumped to output file
+        """
+        args = {
+            'token': self.access_token,
+            'namespace': namespace,
+            'pod': pod,
+            'container': container,
+            'command': ''.join(['command={}&'.format(word) for word in command.split()])
+        }
+        url = urljoin(
+            self.kubernetes_api,
+            'namespaces/{namespace}/pods/{pod}/exec?'
+            'access_token={token}&container={container}&'
+            '{command}stdout=1&stdin=0&tty=0'.format(**args))
+
+        # The above endpoint needs the request to be upgraded to SPDY,
+        # which python-requests does not yet support. However, the same
+        # endpoint works over websockets, so we are using websocket client.
+
+        # Convert url from http(s) protocol to wss protocol
+        url = 'wss://' + url.split('://', 1)[-1]
+        logger.debug('url: {}'.format(url))
+
+        results = []
+
+        ws = websocket.WebSocketApp(
+            url,
+            on_message=lambda ws, message: self._handle_exec_reply(ws, message, results, outfile))
+
+        ws.run_forever(sslopt={'cert_reqs': ssl.CERT_NONE})
```

Comment:
after #480 is merged we'll need to update this so it uses `self.provider_ca` and `self.provider_tls_verify`